driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -59,6 +61,34 @@ public void setUp() {
         collection.insertOne(Document.parse("{ _id : 0 }"));
     }
 
+    class ChangeStreamWatchThread extends Thread {
```

Comment:
This class should implement Runnable, not extend Thread. Especially as you are passing an instance of it to the Thread constructor.  And you should rename to ChangeStreamWatcher in that case

Reply:
The class is now named ChangeStreamWatcher and implements Runnable.